api/v1alpha8/openstackcluster_types.go

```diff
@@ -65,10 +65,16 @@ type OpenStackClusterSpec struct {
 	// ExternalRouterIPs is an array of externalIPs on the respective subnets.
 	// This is necessary if the router needs a fixed ip in a specific subnet.
 	ExternalRouterIPs []ExternalRouterIPParam `json:"externalRouterIPs,omitempty"`
-	// ExternalNetworkID is the ID of an external OpenStack Network. This is necessary
-	// to get public internet to the VMs.
+
+	// ExternalNetwork is the OpenStack Network to be used to get public internet to the VMs.
+	// +optional
+	ExternalNetwork NetworkFilter `json:"externalNetwork,omitempty"`
+
+	// DisableExternalNetwork determines whether or not to attempt to connect the cluster
+	// to an external network. This allows for the creation of clusters when connecting
+	// to an external network is not possible or desirable, e.g. if using a provider network.
 	// +optional
-	ExternalNetworkID string `json:"externalNetworkId,omitempty"`
+	DisableExternalNetwork bool `json:"disableExternalNetwork"`
```

Comment:
Question: given `ExternalNetwork` is optional, couldn't we simply suggest that leaving this undefined/explicitly set to `null` indicates that the external network should be disabled?

*Later:* Okay, an unset `externalNetwork` value already carries a special meaning. I wonder if we can change that meaning. I discuss this later.

Reply:
> couldn't we simply suggest that leaving this undefined/explicitly set to null indicates that the external network should be disabled?

I thought about it too but we can't because to remain backward compatible, we have to support that users using the default (not set), CAPO will try to find an external network for them.
If we are ok breaking this then indeed we don't need `DisableExternalNetwork`.